src/Native/Unix/System.Native/pal_networking.cpp

```diff
@@ -2050,42 +2058,34 @@ extern "C" Error SystemNative_GetSockOpt(
     //
     if (socketOptionLevel == PAL_SOL_SOCKET)
     {
-        if (socketOptionName == PAL_SO_EXCLUSIVEADDRUSE)
+        if (socketOptionName == PAL_SO_EXCLUSIVEADDRUSE || socketOptionName == PAL_SO_REUSEADDR)
         {
-            //
-            // SO_EXCLUSIVEADDRUSE makes Windows behave like Unix platforms do WRT the SO_REUSEADDR option.
-            // So, for non-Windows platforms, we act as if SO_EXCLUSIVEADDRUSE is always enabled.
-            //
             if (*optionLen != sizeof(int32_t))
             {
                 return PAL_EINVAL;
             }
 
-            *reinterpret_cast<int32_t*>(optionValue) = 1;
-            return PAL_SUCCESS;
-        }
-        else if (socketOptionName == PAL_SO_REUSEADDR)
-        {
-            //
-            // On Windows, SO_REUSEADDR allows the address *and* port to be reused.  It's equivalent to 
-            // SO_REUSEADDR + SO_REUSEPORT other systems.  Se we only return "true" if both of those options are true.
-            //
             auto optLen = static_cast<socklen_t>(*optionLen);
+            // On Unix, SO_REUSEPORT controls the ability to bind multiple sockets to the same address.
+            int err = getsockopt(fd, SOL_SOCKET, SO_REUSEPORT, optionValue, &optLen);
 
-            int err = getsockopt(fd, SOL_SOCKET, SO_REUSEADDR, optionValue, &optLen);
-
-            if (err == 0 && *reinterpret_cast<uint32_t*>(optionValue) != 0)
+            if (err != 0)
             {
-                err = getsockopt(fd, SOL_SOCKET, SO_REUSEPORT, optionValue, &optLen);
+                return SystemNative_ConvertErrorPlatformToPal(errno);
             }
 
-            if (err != 0)
+            int value = *reinterpret_cast<int32_t*>(optionValue);
+
+            // macOS returns non-zero values other than 1.
+            value = value == 0 ? 0 : 1;
```

Comment:
Sorry, I just noticed this. Should this be in an else for the if block below?

Reply:
(Can be addressed in a subsequent cleanup PR if you agree, along with my other nit.)